eureka-client/src/main/java/com/netflix/discovery/shared/Applications.java

```diff
@@ -186,25 +175,18 @@ public Application getRegisteredApplications(String appName) {
      * @return list of <em>instances</em>.
      */
     public List<InstanceInfo> getInstancesBySecureVirtualHostName(String secureVirtualHostName) {
-        AtomicReference<List<InstanceInfo>> ref = this.shuffledSecureVirtualHostNameMap
-                .get(secureVirtualHostName.toUpperCase(Locale.ROOT));
-        if (ref == null || ref.get() == null) {
-            return new ArrayList<InstanceInfo>();
-        } else {
-            return ref.get();
-        }
+        return Optional.ofNullable(this.secureVirtualHostNameAppMap.get(secureVirtualHostName.toUpperCase(Locale.ROOT)))
```

Comment:
Sorry I should have been more clear. I was thinking of something along the lines of 
```
private List<InstanceInfo> static internalGetInstancesByVirtualHostName(Map<> vipAppMapping, String vipName) {
    return Optional.ofNullable(vipAppMapping.get(vipName.toUpperCase(Locale.ROOT)))
    .map(VipIndexSupport::getVipList)
     .map(AtomicReference::get)
     .orElseGet(Collections::emptyList); 
}
```